modules/editor-events/assets/js/editor/module.js

```diff
@@ -0,0 +1,48 @@
+import EventsStorage from './events-storage';
+
+export default class extends elementorModules.Module {
+	types = {
+		click: 'click',
+	};
+
+	events = {
+		site_settings: 'site_settings',
+	};
+
+	onInit() {
+		if ( ! elementor.config.editor_events?.can_send_events ) {
+			return;
+		}
+
+		window.addEventListener( 'unload', this.sendEvents() );
+	}
+
+	dispatchEvent( type, eventId, context ) {
+		if ( ! elementor.config.editor_events?.can_send_events ) {
+			return;
+		}
+
+		EventsStorage.set( { type, event_id: eventId, context, timestamp: Date.now() } );
+	}
+
+	sendEvents() {
+		const events = EventsStorage.get();
+
+		if ( ! events.length ) {
+			return;
+		}
+
+		EventsStorage.clear();
```

Comment:
Maybe we can use an 'optimistic approach' while saving the data in case it doesn't get sent correctly, and then in the catch we can store it back to be sent with the next batch? Does it make sense?

Reply:
good idea, done